Realm/RLMResults.h

```diff
@@ -408,6 +427,169 @@ __attribute__((warn_unused_result));
                                                          NSError *_Nullable error))block
                                       keyPaths:(nullable NSArray<NSString *> *)keyPaths
 __attribute__((warn_unused_result));
+
+#pragma mark - Flexible Sync
+
+/**
+ Creates an RLMSyncSubscription matching the RLMResults' local filter.
+
+ After committing the subscription to the realm's local subscription set, the method
+ will wait for downloads according to ``RLMWaitForSyncModeOnCreation`` behavior.
+ @see ``RLMWaitForSyncModeOnCreation`` and ``RLMWaitForSyncMode``
+
+ If `subscribeWithCompletion` is called on a query
+ that's already subscribed to without a name, another subscription is not created.
+ If `subscribeWithCompletion` is called on a query
+ that's already subscribed to with a name, an additional subscription is created without a name.
+
+ This method opens a write transaction that creates or updates a subscription.
+ It's advised to *not* loop over this method in order to create multiple subscriptions at once.
+ This could create a performance bottleneck by opening multiple unnecessary write transactions.
+ @see: `[RLMSyncSubscription update:queue:onComplete:]` in order to create multiple subscriptions.
+
+ @param completion The block called after the subscription download
+ is completed. The completion is called according to
+ RLMWaitForSyncModeOnCreation behavior.
+ @see ``RLMWaitForSyncModeOnCreation`` and ``RLMWaitForSyncMode``
+ @param queue The queue where the completion disptaches.
+ */
+- (void)subscribeWithCompletion:(RLMResultsCompletionBlock)completion
```

Comment:
This should follow the convention of the other API's where the queue comes before the block. So it would look something like:
```
- (void)subscribeWithCompletionOnQueue:(dispatch_queue_t _Nullable)queue 
                                             completionBlock:(RLMResultsCompletionBlock)completion;